### PR TITLE
[SYCL] Fix kernel attribute codegen

### PR DESCRIPTION
**Pull request by MrSidims**
It was failed to process correctly a case of multiple attrs
attached to SYCL kernel.

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

File: clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// RUN: %clang_cc1 -std=c++11 -triple spir64-unknown-linux-sycldevice -disable-llvm-passes -fsycl-is-device -emit-llvm -o - %s | FileCheck %s
```
Comment (AGindinson) [inline clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp]:
```suggestion
// RUN: %clang_cc1 -std=c++11 -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -fsycl-is-device -emit-llvm -o - %s | FileCheck %s
```

File: clang/lib/Sema/SemaSYCL.cpp [hunk 2]
```diff
@@ -426,9 +426,9 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
 
       if (auto *A = FD->getAttr<IntelReqdSubGroupSizeAttr>())
         Attrs.insert(A);
-      else if (auto *A = FD->getAttr<ReqdWorkGroupSizeAttr>())
+      if (auto *A = FD->getAttr<ReqdWorkGroupSizeAttr>())
         Attrs.insert(A);
```
Comment (Fznamznon) [inline clang/lib/Sema/SemaSYCL.cpp]: Suggest merging these `if` statements and adding early exit.

File: clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+class Functor {
+public:
+  [[cl::intel_reqd_sub_group_size(4), cl::reqd_work_group_size(32, 16, 16)]] void operator()() {}
```
Comment (Fznamznon) [inline clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp]: Can these attribute conflict with each other?
If yes, we should make sure that the diagnostic is correct in this case.

File: clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+};
+
+
+template <typename name, typename Func>
```
Comment (Fznamznon) [inline clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp]:
```suggestion
template <typename Name, typename Func>
```

File: clang/lib/Sema/SemaSYCL.cpp (hunk 2 above)
Comment (MrSidims) [reply clang/lib/Sema/SemaSYCL.cpp]: I can merge these ifs, but why early exit?

Comment (Fznamznon) [reply clang/lib/Sema/SemaSYCL.cpp]: No reason. Never mind :)

File: clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp (hunk 3 above)
Comment (MrSidims) [reply clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp]: I don't think they do.

File: clang/lib/Sema/SemaSYCL.cpp (hunk 2 above)
Comment (MrSidims) [reply clang/lib/Sema/SemaSYCL.cpp]: Also I think that these ifs aren't really mergeable.

File: clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp (hunk 1 above)
Comment (MrSidims) [reply clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp]: Done

File: clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp (hunk 4 above)
Comment (MrSidims) [reply clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp]: Done

File: clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp (hunk 3 above)
Comment (MrSidims) [reply clang/test/CodeGenSYCL/sycl-multi-kernel-attr.cpp]: But these checks will be added in https://github.com/intel/llvm/pull/883 for another pair of attributes.

File: clang/lib/Sema/SemaSYCL.cpp (hunk 2 above)
Comment (Fznamznon) [reply clang/lib/Sema/SemaSYCL.cpp]: Probably you are right.

**Pull request merged**